consoles/sshVirtsh.pm

```diff
@@ -489,48 +487,93 @@ sub define_and_start ($self) {
         $self->run_cmd(
             "cat > $libvirtauthfilename <<__END
 [credentials-vmware]
-username=" . ($bmwqemu::vars{VMWARE_USERNAME} or die 'Need variable VMWARE_USERNAME') . "
-password=" . ($bmwqemu::vars{VMWARE_PASSWORD} or die 'Need variable VMWARE_PASSWORD') . "
-[auth-esx-" . ($bmwqemu::vars{VMWARE_HOST} or die 'Need variable VMWARE_HOST') . "]
+username=$self->{vmm_user}
```

Comment:
Here you're using hash keys but before these values are stored using attributes. That seems inconsistent and is maybe also wrong.

Reply:
Not sure why, but then in is interpreted as this.

```
  username=consoles::sshVirtsh=HASH(0x557622de7e78)->vmm_user
  password=consoles::sshVirtsh=HASH(0x557622de7e78)->vmm_pword
  [auth-esx-consoles::sshVirtsh=HASH(0x557622de7e78)->vmm_host]
```

Reply:
Or should it be like this ?
```
...
username=${self->vmm_user}
...
```

Reply:
To my knowledge you cannot call a function inside a string literal (and an attribute is basically a function). You can store the values in variables before making the string. You can also simply *not* using attributes (consistently). Especially if these fields should only be used by the object internally, it would make sense to use hash-keys with leading underscore like `$self->{_vmm_user}` (which btw. also gives better performance than attributes).

Reply:
Updated, thanks!